README.md

```diff
@@ -7,185 +7,181 @@
-     with
-       Pre => (Is_Contained (Buffer) and then Valid_Value_Length (Buffer));
-
-   function Valid_Value (Buffer : Types.Bytes) return Boolean
-     with
-       Pre => Is_Contained (Buffer);
-
-   function Value_First (Buffer : Types.Bytes) return Types.Index_Type
-     with
-       Pre => (Is_Contained (Buffer) and then Valid_Value (Buffer));
-
-   function Value_Last (Buffer : Types.Bytes) return Types.Index_Type
-     with
-       Pre => (Is_Contained (Buffer) and then Valid_Value (Buffer));
-
-   procedure Value (Buffer : Types.Bytes; First : out Types.Index_Type; Last : out Types.Index_Type)
-     with
-       Pre => (Is_Contained (Buffer) and then Valid_Value (Buffer)),
-       Post => (First = Value_First (Buffer) and then Last = Value_Last (Buffer));
-
-   function Is_Valid (Buffer : Types.Bytes) return Boolean
-     with
-       Pre => Is_Contained (Buffer);
-
-end Foo.Bar;
+$ rflx generate specs/tlv.rflx generated
+Parsing specs/tlv.rflx... OK
+Generating... OK
+Created generated/rflx-tlv.ads
+Created generated/rflx-tlv-generic_message.ads
+Created generated/rflx-tlv-generic_message.adb
+Created generated/rflx-tlv-message.ads
+Created generated/rflx.ads
+Created generated/rflx-types.ads
+Created generated/rflx-types.adb
+Created generated/rflx-message_sequence.ads
+Created generated/rflx-message_sequence.adb
+Created generated/rflx-scalar_sequence.ads
+Created generated/rflx-scalar_sequence.adb
 ```
 
-The check of the `Is_Contained` predicate in the precondition of each access and validation function ensures that always the right input buffer is used. This is important if multiple protocol layers or [type refinements](/doc/Language-Reference.md#type-refinement) are used. At the lowest protocol layer the predicate has to be added manually to the input buffer by calling `Label`.
-
-The generated code could be used in the following way:
+### Use of Generated Code
+
+All atomic types defined in the specification are represented by a similar Ada type in the generated code. For `TLV` the following types are defined in the package `RFLX.TLV`:
+
+- `type Tag_Type is (Msg_Data, Msg_Error) with Size => 2`
+- `for Tag_Type use (Msg_Data => 1, Msg_Error => 3);`
+- `type Length_Type is mod 2**14`
+
+All types and subprograms related to `Message` can be found in the package `RFLX.TLV.Message`:
+
+- `type Context_Type`
+    - stores buffer and internal state
+- `function Create return Context_Type`
+    - return default initialized context
+- `procedure Initialize (Context : out Context_Type; Buffer : in out RFLX.Types.Bytes_Ptr)`
+    - initialize context with buffer
+- `procedure Initialize (Context : out Context_Type; Buffer : in out RFLX.Types.Bytes_Ptr; First, Last : RFLX.Types.Bit_Index_Type)`
+    - initialize context with buffer and explicit bounds
+- `procedure Take_Buffer (Context : in out Context_Type; Buffer : out RFLX.Types.Bytes_Ptr)`
+    - take buffer from context
```

Comment:
What does "take" mean -> We should explain what's happening and what the implications are...